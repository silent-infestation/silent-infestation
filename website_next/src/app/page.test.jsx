// ./page.test.jsx
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Index from './page';

// Mock du composant Contact
jest.mock('@/components/Contact', () => {
  return function MockContact() {
    return <div data-testid="mock-contact">Contact Component</div>;
  };
});

describe.skip('Index Page', () => {
  it('rend correctement la page', () => {
    render(<Index />);
    expect(screen.getByTestId('mock-contact')).toBeInTheDocument();
  });

  it('rend le composant Contact', () => {
    render(<Index />);
    const contactComponent = screen.getByTestId('mock-contact');
    expect(contactComponent).toBeInTheDocument();
    expect(contactComponent).toHaveTextContent('Contact Component');
  });

  it('maintient la structure attendue de la page', () => {
    const { container } = render(<Index />);
    expect(container.firstChild?.childNodes).toHaveLength(1);
  });

  it('correspond au snapshot', () => {
    const { container } = render(<Index />);
    expect(container).toMatchSnapshot();
  });

  it('intÃ¨gre correctement avec le composant Contact', () => {
    const { container } = render(<Index />);
    const contactElement = screen.getByTestId('mock-contact');
    expect(contactElement).toBeInTheDocument();
    expect(container.firstChild).toContainElement(contactElement);
  });
});
