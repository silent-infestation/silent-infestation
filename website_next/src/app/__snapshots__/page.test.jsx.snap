// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page correspond au snapshot 1`] = `
<div>
  <div
    data-testid="mock-contact"
  >
    Contact Component
  </div>
</div>
`;
